microcosm_pubsub/daemon.py

```diff
@@ -46,27 +54,36 @@ def process(self):
         """
         Lambda Function method that runs only once
         """
-        self.initialize()
         self.graph.logger.info("Local starting daemon {}".format(self.name))
         with self.graph.error_policy:
             self.graph.sqs_message_dispatcher.handle_batch(self.bound_handlers)
 
     @classmethod
     def make_lambda_handler(cls):
+        # initializing graph object during first call, and any subsequent calls would reuse execution context
+        # unless the execution context is unavailable
+        daemon = cls(enable_lambda_mode=True)
+        daemon.initialize()
+
         def handler(event, context):
             """
             AWS Lambda function handler.
             """
-            # this is for the warmup event.
-            # just return something and don't continue
+            # warmp events generated via Cloudwatch to keep Lambda's warm, which means that past invocation's
+            # execution context would be used again. If Lambda's become cold, that container startup time and
+            # microcosm graph loading time would slow the message processing
             if "warm" in event:
+                daemon.graph.logger.debug("Processing warm-up cloudwatch event")
                 return "warming up"
 
             # we configure SQS Queue to use batches of 1,
             # so received event contains
             # another stringified json with actual message inside
-            daemon = cls(event=event["Records"][0])
+            daemon.graph.logger.debug("Processing SQS event: {}".format(event["Records"][0]))
```

Comment:
@pankajghosh do we want this debug log line here? it also violates some [indirection principles](https://en.wikipedia.org/wiki/Law_of_Demeter) given how deep it goes into the object graph..